accname/aria-owns.html

```diff
@@ -0,0 +1,174 @@
+<!doctype html>
+<html>
+
+<head>
+  <meta charset="utf-8">
+  <title>Accname tests for assorted combinations of aria-owns, aria-hidden, and host language hiding techniques</title>
+  <script src="/resources/testharness.js"></script>
+  <script src="/resources/testharnessreport.js"></script>
+  <script src="/resources/testdriver.js"></script>
+  <script src="/resources/testdriver-vendor.js"></script>
+  <script src="/resources/testdriver-actions.js"></script>
+  <script src="/wai-aria/scripts/aria-utils.js"></script>
+  <style>
+    [aria-hidden="true"] {
+      outline: 3px dashed darkred;
+    }
+
+    [id] {
+      outline: 1px dashed green;
+    }
+  </style>
+</head>
+
+<body>
+
+  <h1>Accname tests for assorted combinations of <code>aria-owns</code>, <code>aria-hidden</code>, and host language hiding techniques</h1>
+  <p><span aria-hidden="true">Text enclosed in a dashed red outline</span> is <code>aria-hidden="true"</code>.</p>
+  <p><span id="foo">Bold text with a yellow background</span> is the target of another element’s <code>aria-owns</code>.</p>
+
+  <h2>A button using aria-owns to specify its label</h2>
+  <button
+    class="ex-label"
+    data-testname="A button using aria-owns to specify its label"
+    data-expectedlabel="Play"
+    aria-owns="play"
+  >
+    <div aria-hidden="true">
+      <span id="play">Play</span>
+      <span id="pause" class="hidden-visually">Pause</span>
+    </div>
+  </button>
+
+  <h2>A link using aria-owns to concatenate extra text</h2>
+  <a
+    class="ex-label"
+    data-testname="A link using aria-owns to concatenate extra text"
+    data-expectedlabel="World Wide Web Consortium (opens in a new window)"
+    href="https://www.w3.org/"
+    target="_blank"
+    aria-owns="new-window-warning-1"
+  >
+    World Wide Web Consortium
+  </a>
+  <div aria-hidden="true" class="hidden-visually">
+    <span id="new-window-warning-1"> (opens in a new window)</span>
+  </div>
+
+  <!--
+    Re: https://github.com/w3c/aria/pull/1839
+    “User agents MUST NOT resolve aria-owns when it is set on an element that
+    has been excluded from the accessibility tree.”
+  -->
+  <h2>A hidden element failing to adopt text using aria-owns</h2>
+  <h3
+    class="ex-label"
+    data-testname="A hidden element failing to adopt text using aria-owns"
+    data-expectedlabel="treasure"
+  >
+    <span hidden aria-owns="treasure">
+      This ninja can’t steal the...
+    </span>
+    <mark id="treasure">treasure</mark>
+  </h3>
+
+  <!--
+    Re: https://github.com/w3c/aria/pull/1839
+    “User agents MUST NOT resolve aria-owns when it is set on an element that
+    has been excluded from the accessibility tree.”
+  -->
+  <h2>A hidden element failing to adopt text from a content-named element using aria-owns</h2>
+  <h3
+    class="ex-label"
+    data-testname="A hidden element failing to adopt text from a content-named element using aria-owns"
```

Comment:
likewise something like "ignore aria-owns when on a hidden element further down in a namefrom: content traversal"